src/main/java/net/dries007/tfc/common/blockentities/rotation/CreativeRotationBlockEntity.java

```diff
@@ -0,0 +1,150 @@
+    {
+        motor.checkForLastTickSync();
+
+        clientTick(level, pos, state, motor);
+    }
+
+    public static void clientTick(Level level, BlockPos pos, BlockState state, CreativeRotationBlockEntity motor)
+    {
+        final Rotation.Tickable rotation = motor.node.rotation();
+        rotation.tick();
+        float target = Mth.abs(motor.step * LERP_SPEED);
+        if (rotation.speed() != target)
+        {
+            rotation.setSpeed(target);
+        }
+    }
+
+    private final CreativeSourceNode node;
+    private int step;
+
+    public CreativeRotationBlockEntity(BlockPos pos, BlockState state)
+    {
+        this(TFCBlockEntities.CREATIVE_MOTOR.get(), pos, state);
+    }
+
+    protected CreativeRotationBlockEntity(BlockEntityType<?> type, BlockPos pos, BlockState state)
+    {
+        super(type, pos, state);
+        Direction.Axis axis = state.getValue(CreativeRotationBlock.AXIS);
+        this.node = new CreativeSourceNode(pos, Node.ofAxis(axis), Direction.fromAxisAndDirection(axis, Direction.AxisDirection.POSITIVE), 0f)
+        {
+            @Override
+            public String toString()
+            {
+                return "CreativeRotator[pos=%s, axis=%s]".formatted(pos(), axis);
+            }
+        };
+    }
+
+    @Override
+    public void markAsInvalidInNetwork()
+    {
+        // Should this do anything?
+    }
+
+    @Override
+    public boolean isInvalidInNetwork()
+    {
+        return false;
+    }
+
+    @Override
+    public Node getRotationNode()
+    {
+        return node;
+    }
+
+    @Override
+    protected void saveAdditional(CompoundTag tag, HolderLookup.Provider provider)
+    {
+        super.saveAdditional(tag, provider);
+        tag.putInt("step", step);
+    }
+
+    @Override
+    protected void loadAdditional(CompoundTag tag, HolderLookup.Provider provider)
+    {
+        super.loadAdditional(tag, provider);
+        step = tag.getInt("step");
+    }
+
+    @Override
+    protected void onLoadAdditional()
+    {
+        performNetworkAction(NetworkAction.ADD_SOURCE);
+    }
+
+    @Override
+    protected void onUnloadAdditional()
+    {
+        performNetworkAction(NetworkAction.REMOVE);
+    }
+
+    public void incrementSpeed()
+    {
+        step = Mth.clamp(step + 1, -MAX_STEPS, MAX_STEPS);
+        if (step == 1)
+        {
+            node.setDirection(getBlockState().getValue(CreativeRotationBlock.AXIS), Direction.AxisDirection.POSITIVE);
+            performNetworkAction(NetworkAction.UPDATE);
+        }
+    }
+
+    public void decrementSpeed()
+    {
+        step = Mth.clamp(step - 1, -MAX_STEPS, MAX_STEPS);
+        if (step == -1)
+        {
+            node.setDirection(getBlockState().getValue(CreativeRotationBlock.AXIS), Direction.AxisDirection.NEGATIVE);
+            performNetworkAction(NetworkAction.UPDATE);
+        }
+    }
+
+    private static class CreativeSourceNode extends SourceNode
```

Comment:
Am i missing a reason why we are instantiating this as an anonymous class above and then still having a static definition of it below? do one or the other not both

Reply:
This was just a case of using a local variable in the toString and got left behind in a refactor